tests/templates/test_state_preparations.py

```diff
@@ -359,6 +390,109 @@ def circuit(state_vector):
             state_vector = np.array([0, 2, 0, 0])
             circuit(state_vector)
 
+    # fmt: off
+    @pytest.mark.parametrize("state_vector, n_wires", [
+        ([.5,.5,.5,.5],2),
+        ([1, 0, 0, 0], 2),
+        ([0, 1, 0, 0], 2),
+        ([0, 0, 0, 1], 2),
+        ([0, 1, 0, 0, 0, 0, 0, 0], 3),
+        ([0, 0, 0, 0, 1, 0, 0, 0], 3),
+        ([2/3, 0, 0, 0, 1/3, 0, 0, 2/3], 3),
+        ([1/2, 0, 0, 0, 1/2, 1/2, 1/2, 0], 3),
+        ([1/3, 0, 0, 0, 2/3, 2/3, 0, 0], 3),
+        ([2/3, 0, 0, 0, 1/3, 0, 0, 2/3], 3),
+    ])
+    # fmt: on
+    def test_RZ_skipped(self, state_vector, n_wires):
+        """Tests whether the cascade of RZ gates is skipped.
+        Note that currently there are also inefficiencies in the number of CNOT gates in the RY cascade.
+        When these are updated, this test will become ineffective"""
+       
+        n_CNOT = 2**n_wires - 2
+
+        dev = qml.device("default.qubit", wires=n_wires)
+
+        @qml.qnode(dev)
+        def circuit(state_vector):
+            MottonenStatePreparation(state_vector, wires=range(n_wires))
+            return qml.expval(qml.PauliX(wires=0))
+
+        # when the RZ cascade is skipped, CNOT gates should only be those required for RY cascade
+        circuit(state_vector)
+
+        assert circuit.qtape.get_resources()["CNOT"] <= n_CNOT
```

Comment:
```suggestion
        assert circuit.qtape.get_resources()["CNOT"] == n_CNOT
```
I personally preferred the equality. The idea isn't that this test should never fail - it's ok if it fails if someone changes `Mottonen...` later on - it'll be a nice confirmation that something has changed and will help us understand and update the tests accordingly.